src/hint_deploy.py

```diff
@@ -190,30 +206,44 @@ def hint_start(obj, cfg, args):
         print("Adding test user '{}'".format(email))
         hint_user(cfg, "add-user", email, pull, "password")
 
+    loadbalancer_configure(obj)
+
 
 def hint_upgrade_hintr(obj):
-    hintr = obj.containers.find("hintr")
+    loadbalancer = obj.containers.find("hintr")
+    hintr_api = obj.containers.find("hintr_api")
     calibrate_worker = obj.containers.find("calibrate_worker")
     worker = obj.containers.find("worker")
-    container = hintr.get(obj.prefix)
+    hintr_containers = hintr_api.get(obj.prefix)
+    loadbalancer_container = loadbalancer.get(obj.prefix)
 
     # Always pull the docker image - and do this *before* we start
     # removing things to minimise downtime.
-    docker_util.image_pull(hintr.name, str(obj.data.hintr_ref))
-    docker_util.image_pull(hintr.name, str(obj.data.hintr_worker_ref))
-
-    if container:
-        if container.status == "running":
-            print("Stopping previous hintr and workers")
-            container.exec_run(["hintr_stop"])
-        docker_util.container_remove_wait(container)
-
-    obj.start(subset=[hintr.name, calibrate_worker.name, worker.name])
+    docker_util.image_pull(loadbalancer.name, str(
+        obj.data.hintr_loadbalancer_ref))
+    docker_util.image_pull(hintr_api.name, str(obj.data.hintr_ref))
+    docker_util.image_pull(hintr_api.name, str(obj.data.hintr_worker_ref))
+
+    for container in hintr_containers:
+        if container:
+            if container.status == "running":
+                print("Stopping {}".format(container.name))
+                container.exec_run(["hintr_stop"])
+            docker_util.container_remove_wait(container)
+    print("Stopping {}".format(loadbalancer_container.name))
+    docker_util.container_stop(
+        loadbalancer_container, True, loadbalancer_container.name)
```

Comment:
So the loadbalancer takes a long time to stop > 3s consistently which is causing `container_stop` to exit early thinking something has gone wrong. So I have just put `kill=True` here temporarily to avoid this problem. I think ideally we can put a timeout on the container for how long it will wait for it to stop or use an approach like the hintr api container and run a script to stop the running container.

Reply:
We could leave this as a kill, the downside would be if we did `./hintr stop` it will fail when trying to stop the load balancer

Reply:
we discussed this and now kill both here and in stop